hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/HoodieROTablePathFilter.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.hoodie.hadoop;
+
+import com.uber.hoodie.common.model.HoodieTableMetadata;
+import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.exception.InvalidDatasetException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * Given a path is a part of
+ * - Hoodie dataset => accepts ONLY the latest version of each path
+ * - Non-Hoodie dataset => then always accept
+ *
+ * We can set this filter, on a query engine's Hadoop Config & if it respects path filters, then
+ * you should be able to query both hoodie & non-hoodie datasets as you would normally do.
+ *
+ * hadoopConf.setClass("mapreduce.input.pathFilter.class",
+ *                      com.uber.hoodie.hadoop.HoodieROTablePathFilter.class,
+ *                      org.apache.hadoop.fs.PathFilter.class)
```

Comment:
We prolly should have documentation on when to use this vs HoodieInputFormat and also the fact that this solution is spark specific.